### ISSUE TITLE
[Enhancement] javadoc and sources jars not included in maven publications

### ISSUE THREAD

**Issue by mch2** [opensearch-project/OpenSearch#1028]
A requirement of publishing to maven central is to include javadoc and sources jars for java publications.

There are javadoc and sources jar tasks defined with the OpenSearch publish plugin [here](https://github.com/opensearch-project/OpenSearch/blob/main/buildSrc/src/main/java/org/opensearch/gradle/PublishPlugin.java#L146).

When running publish commands these artifacts should be build and included with each publication.  Instead these tasks if run separately place them under src/*.
 
**To Reproduce**
1. Clone the repository
2. Execute publishToMavenLocal command
3. Under ~/.m2/repositories/org/opensearch/** See that only .pom, .jar and .module files are included for each.

**Expected behavior**
These artifacts should be included for any jar publication.

**Additional context**
These commands could obviously be executed and then the artifacts copied over from each dir manually, but they should be built & included with a single publish command.

**Comment by mch2** [comment]
A block like this should solve the issue for each publication that needs to end up in central

```
publishing {
  publications {
    nebula {
      artifactId 'opensearch'
      artifact javadocJar
      artifact sourcesJar
    }
  }
}
```

but it would be great to centralize this for anything using the publish plugin vs duplicating across config.

**Comment by frotsch** [comment]
See also https://discuss.opendistrocommunity.dev/t/how-to-build-opensearch-including-all-plugins/6274/4

**Comment by adnapibar** [comment]
Resolved by #1049
